src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let currentState = { ...state };
+
+  for (const operation of actions) {
+    let newState = { ...currentState };
+
+    switch (operation.type) {
+      case 'addProperties':
+        // Добавляем новые свойства
+        newState = { ...newState, ...operation.extraData };
+        break;
+
+      case 'removeProperties':
+        // Удаляем указанные свойства
+        for (const key of operation.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+
+      default:
+        throw new Error(`Unhandled action type: ${operation.type}`);
+    }
+    currentState = newState;
+
+    // Добавляем копию нового состояния в результат
+    result.push(newState);
```

Comment:
The copy of the object should be pushed to the result array at the end of each loop cycle, but outside of the `switch` block, after the current action is processed. This ensures that the state is correctly updated and stored after each action.